api/nakadi-event-bus-api.yaml

```diff
@@ -606,22 +488,389 @@ paths:
           description: Partition id
           type: string
           required: true
-        - name: X-Flow-Id
-          in: header
+        - $ref: '#/parameters/XFlowId'
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/Partition'
+        '401':
+          description: Client is not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+
+  /subscriptions:
+    post:
+      tags:
+        - managed-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        This endpoint creates a subscription for EventTypes. The subscription is needed to be able to
+        consume events from EventTypes in a high level way when Nakadi stores the offsets and manages the
+        rebalancing of consuming clients.
+        The subscription is identified by its key parameters (owning_application, event_types, consumer_group). If
```

Comment:
Well there is one aspect of PUT that doesn't fit well with the current approach. PUT means the client chooses the URL, e.g. if I use `PUT /users/whiskeysierra` I am in full control of the URL and the server must (?) use that one. I think the *correct* (double air quotes!) approach would be to use:

```http
PUT /subscriptions?event_types=foo&owning_application_id=bar
```

Because this looks rather ugly I'd also be ok with POST. It's an interesting problem/challenge. Feel free to ask in the api guild chat for some other ideas/opinions.